File: src/main/java/io/github/alexiscomete/lapinousecond/worlds/buildings/interactions/Banque.java
package io.github.alexiscomete.lapinousecond.worlds.buildings.interactions;

import io.github.alexiscomete.lapinousecond.entity.Player;
import io.github.alexiscomete.lapinousecond.worlds.buildings.Building;
import io.github.alexiscomete.lapinousecond.worlds.buildings.BuildingInteraction;
import org.javacord.api.entity.message.MessageBuilder;
import org.javacord.api.entity.message.embed.EmbedBuilder;

public class Banque extends BuildingInteraction {
    public Banque(Building building) {
        super(building);
    }

    @Override
    public EmbedBuilder getInfos(Player p) {
        return null;
    }

    @Override
    public MessageBuilder getCompleteInfos(Player p) {
        return null;
    }

    @Override
    public void configBuilding() {

    }
}

File: src/main/java/io/github/alexiscomete/lapinousecond/worlds/buildings/interactions/Banque.kt
package io.github.alexiscomete.lapinousecond.worlds.buildings.interactions

import io.github.alexiscomete.lapinousecond.view.ui.longuis.LongCustomUI
import io.github.alexiscomete.lapinousecond.view.ui.longuis.MenuBuilderUI
import io.github.alexiscomete.lapinousecond.view.ui.playerui.PlayerUI
import io.github.alexiscomete.lapinousecond.worlds.buildings.Building
import io.github.alexiscomete.lapinousecond.worlds.buildings.BuildingInteraction

class Banque(building: Building?) : BuildingInteraction(building!!) {
    override fun interpret(args: Array<String>) {}
    override val help: String?
        get() = null
    override val usage: String?
        get() = null

    override fun getCompleteInfos(ui: PlayerUI): LongCustomUI {
        return MenuBuilderUI(
            "Banque",
            "Vous pouvez déposer ou retirer de l'argent de votre compte bancaire ici.",
            ui
        )
    }

    override fun descriptionShort(): String {
        return "Banque"
    }

    override fun title(): String {
        return "Banque"
    }

    override fun configBuilding() {}
}